src/Features/Core/Portable/ChangeSignature/SignatureChange.cs

```diff
@@ -55,5 +58,66 @@ public SignatureChange(ParameterConfiguration originalConfiguration, ParameterCo
 
         internal SignatureChange WithoutAddedParameters()
             => new SignatureChange(OriginalConfiguration, UpdatedConfiguration.WithoutAddedParameters());
+
+        internal void LogTelemetry()
+        {
+            var originalListOfParameters = OriginalConfiguration.ToListOfParameters();
+            var updatedListOfParameters = UpdatedConfiguration.ToListOfParameters();
+
+            ChangeSignatureLogger.LogTransformationInformation(
+                numOriginalParameters: originalListOfParameters.Length,
+                numParametersAdded: updatedListOfParameters.Count(p => p is AddedParameter),
+                numParametersRemoved: originalListOfParameters.Count(p => !updatedListOfParameters.Contains(p)),
+                anyParametersReordered: AnyParametersReordered(originalListOfParameters, updatedListOfParameters));
+
+            foreach (var addedParameter in updatedListOfParameters.OfType<AddedParameter>())
+            {
+                if (addedParameter.IsRequired)
+                {
+                    ChangeSignatureLogger.LogAddedParameterRequired();
+                }
+
+                if (addedParameter.TypeBinds)
+                {
+                    ChangeSignatureLogger.LogAddedParameterTypeBinds();
+                }
+
+                if (addedParameter.IsCallsiteTodo)
+                {
+                    ChangeSignatureLogger.LogAddedParameter_ValueTODO();
+                }
+                else if (addedParameter.IsCallsiteOmitted)
+                {
+                    ChangeSignatureLogger.LogAddedParameter_ValueOmitted();
+                }
+                else
+                {
+                    if (addedParameter.UseNamedArguments)
+                    {
+                        ChangeSignatureLogger.LogAddedParameter_ValueExplicitNamed();
+                    }
+                    else
+                    {
+                        ChangeSignatureLogger.LogAddedParameter_ValueExplicit();
+                    }
+                }
+            }
+        }
+
+        private bool AnyParametersReordered(ImmutableArray<Parameter> originalListOfParameters, ImmutableArray<Parameter> updatedListOfParameters)
+        {
+            var originalListWithoutRemovedOrAdded = originalListOfParameters.Where(p => updatedListOfParameters.Contains(p)).ToImmutableArray();
```

Comment:
nit: WhereAsArray

Reply:
nvm

---
In reply to: [412436796](https://github.com/dotnet/roslyn/pull/43504#discussion_r412436796) [](ancestors = 412436796)